src/test/ui/pattern/usefulness/slice-patterns.rs

```diff
@@ -0,0 +1,96 @@
+#![feature(slice_patterns)]
```

Comment:
This test feels like 3 baked into one:
- check-pass tests,
- non-exhaustiveness tests
- unreachability tests

I would like to see the passing portions of this tests at least separated out to a different test file.
(In general, passing and failing stuff shouldn't be mixed into the same file.)